### PR TITLE
Add test for form.elements with elements in a shadow tree

### PR DESCRIPTION
**Pull request by hzr**
https://github.com/whatwg/html/pull/2825 needs a test.

### CODE REVIEW

**Comment by hzr** [comment]
CC @domenic

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None



View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17412)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/264702854)

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None



View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17412)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/264702854)

**Comment by w3c-bots** [comment]
# Build PASSED

Started: 2017-08-15 11:12:44
Finished: 2017-08-15 11:21:58



View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17412)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/264702854)

**Comment by hzr** [comment]
Does this belong in https://github.com/w3c/web-platform-tests/tree/master/shadow-dom instead?

**Comment by sideshowbarker** [comment]
w3c-test:mirror

File: html/semantics/forms/the-form-element/form-elements-filter.html [hunk 1]
```diff
@@ -169,4 +174,17 @@
...
+  const shadowRoot1 = document.querySelector("#shadow-1").attachShadow({mode: "open"});
+  const input1 = document.createElement("input");
+  shadowRoot1.appendChild(input1);
+
+  const shadowRoot2 = document.querySelector("#shadow-2").attachShadow({mode: "open"});
+  const input2 = document.createElement("input");
+  input2.setAttribute("form", "form2");
+  shadowRoot2.appendChild(input2);
+
+  assert_equals(document.querySelector("#form2").elements.length, 0);
```
Comment (annevk) [inline html/semantics/forms/the-form-element/form-elements-filter.html]: Can you also assert that input1 and input2.form is null?

**Comment by annevk** [comment]
(I think it's fine for these to be here. I don't think we should have a separate directory long term.)

File: html/semantics/forms/the-form-element/form-elements-filter.html (hunk 1 above)
Comment (hzr) [inline html/semantics/forms/the-form-element/form-elements-filter.html]: Was thinking about it, but wasn't sure if it belonged here or with tests for the `form` attribute. Added now. :)

**Review by annevk — APPROVED** [review]

**Comment by annevk** [comment]
@rniwa @hayatoito FYI, new shadow tree tests.

**Pull request merged**